### PR TITLE
META-4193 response body updated

### PR DESCRIPTION
**Pull request by BurakAltas**
## Change description

> Response body structure of lineage on demand API updated 

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+                ", guidEntityMap=" + guidEntityMap +
+                ", relations=" + relations +
+                ", lineageDirection=" + lineageDirection +
+                ", lineageDepth=" + lineageDepth +
+                '}';
+    }
+
+    @JsonAutoDetect(getterVisibility = PUBLIC_ONLY, setterVisibility = PUBLIC_ONLY, fieldVisibility = NONE)
+    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+    @JsonIgnoreProperties(ignoreUnknown = true, value = {"inputRelationsReachedLimit", "outputRelationsReachedLimit"})
```
Comment (mehtaanshul) [inline intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java:174]: inputRelationsReachedLimit, outputRelationsReachedLimit - no variables with this name

File: intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+public class AtlasLineageOnDemandInfo implements Serializable {
+    private String baseEntityGuid;
+    private LineageDirection lineageDirection;
+    private int lineageDepth;
+    private Map<String, AtlasEntityHeader> guidEntityMap;
+    private Set<LineageRelation> relations;
+    private Set<String>                             visitedEdges;
+    private Set<String>                             skippedEdges;
+    private Map<String, LineageInfoOnDemand>        relationsOnDemand;
+    private Map<String, LineageOnDemandConstraints> lineageOnDemandPayload;
```
Comment (mehtaanshul) [inline intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java:35]: api response should not just have LineageOnDemandConstraints, it should have other items as well from the payload

File: intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java (hunk 1 above)
Comment (BurakAltas) [reply intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java]: addressed

File: intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java (hunk 2 above)
Comment (BurakAltas) [reply intg/src/main/java/org/apache/atlas/model/lineage/AtlasLineageOnDemandInfo.java]: addressed

**Review by mehtaanshul — APPROVED** [review]

**Pull request merged**